packages/lib-core/package.json

```diff
@@ -10,16 +10,26 @@
   },
   "files": [
     "dist/index.js",
-    "dist/index.d.ts"
+    "dist/index.d.ts",
+    "dist/index.ts38.d.ts",
+    "dist/build-meta.json"
   ],
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
+  "typesVersions": {
+    "<4.0": {
```

Comment:
@florkbr Let me know if you prefer `>=3.8 <4.0` instead of simply `<4.0`.

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#version-selection-with-typesversions

TypeScript compiler looks at `typesVersions` first and when no matching entry is found, falls back to `types`.

Since we provide a single `.d.ts` bundle, we resolve `index.d.ts` into a single replacement file (instead of using a pattern).

Reply:
I think since this fix is specifically for OCP until they can upgrade to TS 4, I'd prefer to make the version range as specific as possible (IE `>=3.8 <4.0`). We don't want to encourage use of the legacy types IMO. If the need arises we can always switch back to `<4.0`.

Reply:
Agreed, I'll change this to be `>=3.8 <4.0`.

Reply:
@vojtechszocs I do not see this change in the diff. For now I'll mark this LGTM. Could we please follow up with the change? Thanks!